src/memory/query_memory_control.cpp

```diff
@@ -58,34 +60,50 @@ void QueriesMemoryControl::EraseThreadToTransactionId(const std::thread::id &thr
   accessor.remove(thread_id);
 }
 
-utils::MemoryTracker *QueriesMemoryControl::GetTrackerCurrentThread() {
+void QueriesMemoryControl::TrackAllocOnCurrentThread(size_t size) {
   auto thread_id_to_transaction_id_accessor = thread_id_to_transaction_id.access();
 
   // we might be just constructing mapping between thread id and transaction id
   // so we miss this allocation
   auto thread_id_to_transaction_id_elem = thread_id_to_transaction_id_accessor.find(std::this_thread::get_id());
   if (thread_id_to_transaction_id_elem == thread_id_to_transaction_id_accessor.end()) {
-    return nullptr;
+    return;
   }
 
   auto transaction_id_to_tracker_accessor = transaction_id_to_tracker.access();
   auto transaction_id_to_tracker =
       transaction_id_to_tracker_accessor.find(thread_id_to_transaction_id_elem->transaction_id);
-  return &transaction_id_to_tracker->tracker;
+
+  MG_ASSERT(transaction_id_to_tracker != transaction_id_to_tracker_accessor.end(), "Tracker for thread not found");
```

Comment:
I changed this assert, as it can happen that some allocation occurs between creating the transaction id tracker and mapping the thread to the transaction ID, so we will miss that small allocation. That shouldn't happen, only if we change something in code, but in any case, just to be sure that we don't crash memgraph because of that.